src/Consistency.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE MonadComprehensions #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE DataKinds #-}
+{-# Language RecordWildCards #-}
+
+module Consistency (
+  checkCases
+) where
+
+
+import Prelude hiding (GT, LT)
+
+import Data.List
+import Text.PrettyPrint.ANSI.Leijen hiding ((<$>))
+import System.Exit (exitFailure)
+import System.IO (stdout)
+
+import Syntax
+import Syntax.Annotated
+import SMT
+import Syntax.Untyped (makeIface)
+
+-- TODO this is duplicated in hevm Keccak.hs but not exported
+combine :: [a] -> [(a,a)]
+combine lst = combine' lst []
+  where
+    combine' [] acc = concat acc
+    combine' (x:xs) acc =
+      let xcomb = [ (x, y) | y <- xs] in
+      combine' xs (xcomb:acc)
+
+-- | Checks non-overlapping cases
```

Comment:
Would be cool to have a little comment here (or in the docs) describing the query and why it works